### ISSUE TITLE
Touchstart on input cancels ion-content scrolling

### ISSUE THREAD

**Issue by ashconnell** [driftyco/ionic#1078]
If you have an ion-content element with a tonne of inputs, you cant scroll down because the inputs receive the event and focuses immediately. This happens in chrome, iOS 7.1 and on a nexus 5

Have tried using ion-scroll, ion-content with `overflow-scroll='true'`, with and without labels, nothing seems to work.

Is there a convention here that i don't know about? It seems like a massive game breaker for anyone who wants to use inputs

Latest nightly tested

**Comment by peeter-tomberg** [comment]
@adamdbradley Has this issue reappeared on iOS? I cant seem to scroll a page whenever I touch an input / span.input-label and drag up / down. 

```html
<div class="list">
    <label class="item item-input item-stacked-label item-borderless">
        <span class="input-label">label<strong>extra</strong></span>
        <div class="range">
            <input type="range" min="1" max="100" value="1">
        </div>
    </label>
    <label class="item item-input item-stacked-label item-borderless">
        <span class="input-label">label</span>
        <input type="text">
    </label>
</div>
```

**Comment by gary149** [comment]
Yes, seems I have the problem too

**Comment by bownie** [comment]
Slight twist on this issue I think is using a range input in a modal the input doesn't get the events at all.  I've tried turning off scroll on the ion-content but still no joy.

**Comment by peeter-tomberg** [comment]
@adamdbradley reopen perhaps?

**Comment by bownie** [comment]
Apparently I can't reopen this issue because I didn't raise it.  Perhaps I'll raise a new issue and link to this one if I can.

**Comment by gary149** [comment]
Yes @bownie - cannot find why it's not working... keep searching. If someone can reopen

**Comment by briankasingli** [comment]
Simply remove the attribute => overflow-scroll="true" from ion-content, and you'll be well on your way.

**Comment by briankasingli** [comment]
Simply remove overflow-scroll="true" from ion-content, and you'll be well on your way!

**Comment by tomkaba** [comment]
@briankasingli @adamdbradley What if for some reason I need to use overflow-scroll="true"   ...is there any workaround to make form inputs not receiving events while scrolling? (problem is only in iOS, android works fine)

**Comment by briankasingli** [comment]
@tomkaba after spending 60 hours on this problem, I found a work around that enables you to scroll (ion-content, over-scroll="true") when the input field is focused (selected).

Essentially, the reason why you cant scroll when the input field is focused is because the input field, and targeted container that you want to scroll in lives within another <ion-content>. The first <ion-content> will create some special classes which restricts you from scrolling content when your focused in the input field. 

My code looks something like this =>


<ion-content id="mainContent" scroll="false">
   <form>
      <input type="text" /> 
   </form>

   <ion-content overflow-scroll="true">
      <ion-list>
          <ion-item ng-repeat="users">
           {{users}}
          </ion-item>
      <ion-list>
   </ion-content>
</ion-content>

My Solution:
I've done two things once when the app is loaded.

1. $("#mainContent").removeClass("scroll-content"); // ionic-content generates special classes for the HTML element, one of them is "scroll-content". This CSS class ("scroll-content") restricts you to scroll your ionic-list when your focused in the input field when you have overflow-scroll="true" set in the ionic-list's parent div (<ion-content> "example above").
2. cordova.plugins.Keyboard.disableScroll(true); // once the class is removed, your whole app is scrollable, making it look weird. Lock you app again by calling this function. This function requires a plugin => https://github.com/driftyco/ionic-plugin-keyboard.

This solution worked for me, let me know if it worked for you.

**Comment by Manduro** [comment]
@adamdbradley This is an issue again, but only on iOS and only with native scrolling enabled. It should be reopened.

**Comment by seanislegend** [comment]
Just been tearing my hair out trying to fix this issue. I can confirm what @Manduro says; it only blocks scrolling with a label wrapper, not a div wrapper. This change fixed it for me - hopefully it will for someone else too!

**Comment by dudeofea** [comment]
@Manduro I tried replacing labels with divs, but that didn't do the trick on iOS for me. It's probably since I was embedding the labels / inputs in a popup rather than just in ion-content. replacing the inputs with textareas worked though. Must be something weird going on with the inputs and touch events

**Comment by barocsi** [comment]
+1

**Comment by judezhu** [comment]
+1

**Comment by philipbrack** [comment]
Hello I am using ionic@1.7.14 and cordova@6.1.1.  I am seeing this same issue.  I need the overflow-scroll="true" in order to make the form look OK, however now the user cannot scroll on the input elements and that looks terrible again.  does anyone have a good workaround?


    <ion-content overflow-scroll="true">
        </div>
            <form>

**Comment by briankasingli** [comment]
@philipbrack did you try my solution?

**Comment by shivenigma** [comment]
is this fixed? I am tired of ionic and iOS quirks. sure ionic is a great framework but it have big issues that needs to be addressed especially on ios like scroll, blank screen on input focus, blur.etc.

**Comment by mattleonard** [comment]
+1

**Comment by R1Daneel** [comment]
+1

**Comment by Ross-Rawlins** [comment]
+1

**Comment by Elijax** [comment]
+1 @briankasingli nice answer, exactly what i was looking for, my app was a mess, thank you man

**Comment by briankasingli** [comment]
@Elijax thats great news!

**Comment by jblamber** [comment]
+1 still relevant! Using divs instead of labels, and removing overflow scroll fixed the issue for me on ios.

**Comment by pavanidasari** [comment]
make a replace `IonicModule.forRoot(MyApp)`  in your app.module.ts with
`    IonicModule.forRoot(MyApp,
      {
        // These options are available in ionic-angular@2.0.0-beta.2 and up.
        scrollAssist: false,    // Valid options appear to be [true, false]
        autoFocusAssist: false  // Valid options appear to be ['instant', 'delay', false]

        // http://ionicframework.com/docs/v2/api/config/Config/)
      }
    )`

**Comment by pavanidasari** [comment]
Replace `IonicModule.forRoot(MyApp)` 
in your app.module.ts with

`    IonicModule.forRoot(MyApp,
      {
        // These options are available in ionic-angular@2.0.0-beta.2 and up.
        scrollAssist: false,    // Valid options appear to be [true, false]
        autoFocusAssist: false  // Valid options appear to be ['instant', 'delay', false]

        // http://ionicframework.com/docs/v2/api/config/Config/)
      }
    )`

**Comment by pavanidasari** [comment]
Replace IonicModule.forRoot(MyApp)
in your app.module.ts with

` "IonicModule.forRoot(MyApp,
{
// These options are available in ionic-angular@2.0.0-beta.2 and up.
scrollAssist: false, // Valid options appear to be [true, false]
autoFocusAssist: false // Valid options appear to be ['instant', 'delay', false]

    // http://ionicframework.com/docs/v2/api/config/Config/)
  }
)"`

**Comment by pavanidasari** [comment]
Replace `IonicModule.forRoot(MyApp)`
in your app.module.ts with

IonicModule.forRoot(MyApp,
{
// These options are available in ionic-angular@2.0.0-beta.2 and up.
scrollAssist: false, // Valid options appear to be [true, false]
autoFocusAssist: false // Valid options appear to be ['instant', 'delay', false]

    // http://ionicframework.com/docs/v2/api/config/Config/)
  }
)

**Comment by pavanidasari** [comment]
Replace `IonicModule.forRoot(MyApp)`
in your app.module.ts with

IonicModule.forRoot(MyApp,
{
// These options are available in ionic-angular@2.0.0-beta.2 and up.
scrollAssist: false, // Valid options appear to be [true, false]
autoFocusAssist: false // Valid options appear to be ['instant', 'delay', false]

}
)

**Comment by pavanidasari** [comment]
Replace `IonicModule.forRoot(MyApp)`
in your app.module.ts with

    IonicModule.forRoot(MyApp,
      {
        scrollAssist: false,    // Valid options appear to be [true, false]
        autoFocusAssist: false  // Valid options appear to be ['instant', 'delay', false]
      }
    )
